### ISSUE TITLE
BO - An exception is thrown when adding a query containing a wrong column only with php8

### ISSUE THREAD

**Issue by HanaRebaiQA** [PrestaShop/PrestaShop#31553]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

An exception is thrown when adding a query containing a wrong field only with php8 

**PS8.1.0 with php 8.1 : NOK** :x: 

![image](https://user-images.githubusercontent.com/92912932/221601733-11542067-e42e-40fc-9eee-cc82acae0868.png)

**PS8.0.1 with php 8.1 : NOK** :x: 

![image](https://user-images.githubusercontent.com/92912932/221601746-0ae41e14-3fef-4ff3-93c9-6f8c808ad766.png)

**PS8.1.0 with php 7.4 : OK** :heavy_check_mark: 

![image](https://user-images.githubusercontent.com/92912932/221601671-fdd3fbbf-69bc-41dd-8200-ceb53dae672b.png)

**PS8.0.1 with php 7.4 : OK** :heavy_check_mark: 

![image](https://user-images.githubusercontent.com/92912932/221601703-e8986dbe-0adc-4eb1-948c-240bcfda6109.png)

So, **it's a regression related to php**!

### Expected behavior

An alert message "Bad query" is displayed!

### Steps to reproduce

1. Use PHP 8.1
2. Go to CONFIGURE > Advanced parameters > Database
3. Add new SQL query > add a bad query that contains a wrong column
```
SELECT COUNT(CUSTOMER_ID) AS CustomerCount, hana_customer.firstname AS Name
  FROM hana_customer C
GROUP BY hana_customer.firstname
```
7. Add a name for the SQL query
8. Click on Save
9. See error > An exception is displayed!

### PrestaShop version(s) where the bug happened

8.0.1, 8.1.0

### PHP version(s) where the bug happened

8.1

**Comment by HanaRebaiQA** [comment]
It's related to #31551

**Comment by MatShir** [comment]
Seems to be fixed with this PR https://github.com/PrestaShop/PrestaShop/pull/31667

**Comment by florine2623** [comment]
It is indeed fixed with PS 8.1.0, on branch `develop` 

<img width="1438" alt="Screenshot 2023-04-06 at 14 27 29" src="https://user-images.githubusercontent.com/16019289/230378860-b7478e1a-2a30-4c0d-80bb-442077fd2d7e.png">

I will close this issue!
